Automattic/wp-calypso
# Staging Sites: Disable button to purchase theme

Related to https://github.com/Automattic/dotcom-forge/issues/2042

## Proposed Changes

This pull request effectively disables the purchase functionality for staging sites. As paid themes cannot be acquired for a staging site, given the absence of a corresponding plan, this PR serves to guarantee that such purchases are not possible.

This pr doesn't fix the bug partially. The sync between production ---> staging is not addressed right now

## Testing Instructions



### Test 

#### Test 1
1. Create a new staging site.
2. Navigate to `/themes/<your site>`
3. Search for a paid theme (An example can be found in the pictures below)
4. Ensure you don't see any buttons that allow you to purchase a theme (like shown below).
5. Ensure that by clicking the notice banner you are navigated to the theme page of the production site.

#### Test 2
1. Create a new staging site.
2. Navigate to `/themes/<your site>`
3. Search for a paid theme (An example can be found in the pictures below)
4. Alter `productionSiteForWpcomStaging` to return null. This will make the `productionSiteSlug` to be empty
5. Ensure you don't see any buttons that allow you to purchase a theme (like shown below).
6. Ensure that you see the same banner as before (Test 1), but you are not able to click it.

| **Before** | **After** |
| :---:         |    :---:       |
|![paid_staging_before](https://user-images.githubusercontent.com/497103/229774904-3feca3b2-eb89-43ed-8f0d-00551ace3678.png)|![paid_staging_after](https://user-images.githubusercontent.com/497103/229774974-522c1e26-87eb-413e-88c7-c2934a171cde.png)|



## Pre-merge Checklist
